### ISSUE TITLE
Cannot find module issue

### ISSUE THREAD

**Issue by niftylettuce** [mochajs/mocha#2902]
```
Error: Cannot find module 'has-flag'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/niftylettuce/Public/koa-better-error-handler/node_modules/mocha/node_modules/supports-color/index.js:2:15)
    at Module._compile (module.js:569:30)
    at Module.replacementCompile (/Users/niftylettuce/Public/koa-better-error-handler/node_modules/append-transform/index.js:58:13)
    at Module._extensions..js (module.js:580:10)
    at require.extensions.(anonymous function) (/Users/niftylettuce/Public/koa-better-error-handler/node_modules/babel-register/lib/node.js:152:7)
    at Object.<anonymous> (/Users/niftylettuce/Public/koa-better-error-handler/node_modules/append-transform/index.js:62:4)
```

node v8.0.0
npm v5.0.0

**Comment by ScottFreeCode** [comment]
Seeing as this is a dependency of a dependency, it sounds like a corrupted `node_modules` folder; can you try:
- `npm ls` and see if it warns about `has-flag` being missing
- `npm i mocha` and see if it installs `has-flag` correctly (I tried this after deleting the `node_modules/has-flag` folder manually and it installed `has-flag` even though `mocha` was already installed)
- if that doesn't work, either `npm rm mocha` or delete the `node_modules` folder and then reinstall
- if even that doesn't work, post your OS and version in case there's something going on at that level

**Comment by MrSauceman** [comment]
I have this exact same issue, when I run `npm ls`, I get `npm ERR! missing: has-flag@^1.0.0, required by supports-color@3.1.2`.
When i deleted my `node_modules` directory and reinstalled everything, the issue was still there.
When I ran `npm i mocha`, it installed properly and it worked.
Seems like specifically installing mocha worked, but it does not work when just running `npm i` on a fresh install.
I hope this helps, seems like there is a problem of some sort here, just not sure what it is.

**Comment by ScottFreeCode** [comment]
Huh; sounds like an NPM issue rather than `node_modules` having gotten corrupted. Not only that, but...

> Seems like specifically installing mocha worked, but it does not work when just running  npm i  on a fresh install.

...sounds like an issue with `package-lock.json` (or perhaps `npm-shrinkwrap.json`). There have been a lot of those, although if I recall correctly most are supposed to be fixed in NPM 5.1.0 (which is released, but not yet bundled with a Node version as of 8.1.4).

Do either of you happen to have a `package-lock.json` or `npm-shrinkwrap.json` from immediately before the issue occurred? It might end up being a moot point in NPM >= 5.1, but it'd be nice to have an example of an incorrect lock/shrinkwrap file that causes this -- we may be able to use it to confirm that the issue is present on 5.0.x and fixed in >= 5.1, or if it's still present in >= 5.1 it will surely be helpful for reporting to NPM. (Notably, the fact that more than one person got the issue for `has-flag` specifically seems very suspicious to me, but without having something to dig into I couldn't say whether it means anything.)

**Comment by umakanth-pendyala** [comment]
I have a similar kind of issue. Please take a look at the following link
https://stackoverflow.com/questions/64173341/unable-to-find-server-js-file-while-testing-with-mocha
